libs/langchain/langchain/chains/moderation.py

```diff
@@ -1,11 +1,20 @@
-"""Pass input through a moderation endpoint."""
 from typing import Any, Dict, List, Optional
 
 from langchain_core.pydantic_v1 import root_validator
-
 from langchain.callbacks.manager import CallbackManagerForChainRun
-from langchain.chains.base import Chain
 from langchain.utils import get_from_dict_or_env
+from langchain.chains.base import Chain
+from langchain.utils.openai import is_openai_v1
+
+import openai
+if is_openai_v1():
+    import openai
+    from openai import moderations as moderation
+    from openai.types import ModerationCreateResponse, Moderation
+    from openai.resources.moderations import Moderations
+else:
+    from openai import Moderation as Moderations
+    moderation = Moderations()
 
```

Comment:
unfortunately we can't do imports like this in langchain because of optional dependencies.

If you want to add this to the `libs/partners/openai` langchain_openai package, then you can rely on the v1 client there!